mautic/mautic
# Campaign builder improvements and marketing/transactional emails for campaigns

Please answer the following questions. 

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/77
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation/pull/25
| Issues addressed (#s or URLs) | 
| BC breaks? | N
| Deprecations? | N

### Required
#### Description:

1. This PR revamps our campaign builder. It does away with the right side menu and uses plus signs or buttons on the anchors to inject events. It also provides a way for developers to configure if an action or decision should be restricted from connecting into other actions/decisions (i.e. opens email will only appear in the list if adding to a send email action or dynamic web content action can only be connected into a dynamic web content requested decision along with preventing the dynamic web content action from connecting into the inaction path of a the dwc request decision). Finally the UI redo made the decision action/inaction wording for timing a little clearer. 
![](http://alan.direct/drop/2016-07-30_17-08-16.gif)

2. It adds the concept of transactional/marketing email types to campaign email sends. Transactional will allow a contact to be sent the same email across campaigns multiple times. Marketing emails will check to see if the contact already has received the email (even from another source such as a different campaign, form, etc), and if so not send it again; although the contact continues through the campaign. 
![](http://alan.direct/drop/2016-07-30_23-06-41.png)

3. It adds dev support to assign channel/channel ID to the event log for statistical/display purposes. I.e. being able to query which email triggered specific campaign events. 

#### Steps to test this PR:
1. Run doctrine:migrations:migrate
2. In dev mode (or build prod assets), build new and test old campaigns including some of the scenarios in the description.
3. For transactional/marketing emails, create a new campaign like this:
![](http://alan.direct/drop/2016-07-30_23-05-32.png)
Execute the campaign and open each email sent. Upon opening, the transactional email should be sent again. But the marketing email should not although registered as completed.
